src/world.rs

```diff
@@ -407,11 +407,12 @@ impl<L> LazyInsertInternal for L
 /// component storages have to be borrowed mutably.
 ///
 /// This resource is added to the world by default.
-pub struct LazyInsertions {
-    stack: TreiberStack<Box<LazyInsertInternal>>
+pub struct LazyUpdate {
+    insert_stack: TreiberStack<Box<LazyInsertInternal>>,
```

Comment:
To elaborate this a bit, `LazyInsertInternal` can just access the `World`, so we can just use it for both (after renaming). Just accept entities and a type parameter then, have a private struct with `PhantomData` and implement the removal, please.